app/merginapi.h

```diff
@@ -39,6 +39,7 @@ typedef QList<std::shared_ptr<MerginProject>> ProjectList;
 
 class MerginApi: public QObject {
     Q_OBJECT
+    Q_PROPERTY(QString username READ username WRITE setUsername NOTIFY usernameChanged)
```

Comment:
I don't think username should be writable property with its own signal (authChanged signal should be enough)